File: TOH.cpp
#include <bits/stdc++.h>
using namespace std;

void TOH(int n, char A = 'A', char B = 'B', char C = 'C')
{
	if (n == 1)
	{
		cout << "Move disc " << n << " from " << A << " to " << C << endl;
		return;
	}
	TOH(n - 1, A, C, B);
	cout << "Move disc " << n << " from " << A << " to " << C << endl;
	TOH(n - 1, B, A, C);
}



int main()
{
	int n;
	cin >> n;

	TOH(n);
	return 0;
}

File: Toh.java
import java.util.Scanner;

public class Toh {

	static void TOH(int n, char A, char B, char C) {
		if (n == 0)
			return;
		TOH(n - 1, A, C, B);
		System.out.println("move disc " + n + " from " + A + " --> " + C);
		TOH(n - 1, B, A, C);

	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();

		char A = 'A', B = 'B', C = 'C';

		TOH(n, A, B, C);
	}
}